### ISSUE TITLE
Materialized Views OperationTimedOut and nodes restart after re-create them

### ISSUE THREAD

**Issue by ricardoheyn** [scylladb/scylla#6625]
Hello,

*Installation details*
Scylla version: 3.1.4-0.20200224.c25f627a6
Cluster size: 3
OS: Ubuntu 18.04.4 LTS

I am facing some issues with Materialized Views on Scylla. Following the backtrace logs:

[backtrace.txt](https://github.com/scylladb/scylla/files/4757073/backtrace.txt)

Sometimes I can not load any data from some Materialized Views even limiting the result. It is displaying the follow output:

OperationTimedOut: errors={'xxx.xxx.xxx.xxx': 'Client request timeout. See Session.execute[_async](timeout)'}, last_host=xxx.xxx.xxx.xxx

Usually the issue is happening with the same Materialized Views, and after re-create them some of the nodes restart.

![restarts](https://user-images.githubusercontent.com/10270413/84243776-fcee9580-ab02-11ea-94e5-a69850d50d5e.png)

Let me know how I can help and to avoid this.

Regards,
Ricardo

**Comment by ricardoheyn** [comment]
Hello,

Maybe this backtrace can help as well (seastar - Exceptional future ignored: std::_Nested_exception<no_such_column_family> (Can't find a column family with UUID):


[backtrace2.txt](https://github.com/scylladb/scylla/files/4757169/backtrace2.txt)


Thank you.

Regards,
Ricardo

**Comment by fruch** [comment]
@nyh @psarna, this one rings a bell ?

**Comment by psarna** [comment]
I see that it's from Scylla 3.1, which reached end-of-life a long while ago, so it can be a whole bunch of bugs that got fixed in newer releases... so my first suggestion would be to update and see if the problem persists.

**Comment by slivne** [comment]
@ricardoheyn I am closing this issue if you still hit an issue after upgrade please post and we will reopen and try to help

**Comment by ricardoheyn** [comment]
Hi,

Sorry for taking so long to answer.

Finally I have upgraded the nodes and apparently I found why the issues with Materialized Views. Basically, after upgrade to the version 4.1.2, I have noticed this new log entry in the applications:

`('Unable to complete the operation against any hosts',
{<Host: xxxxxxxxxxxx:9042 xxxxxxxx>: <Error from server: code=0000
[Server error] message="partition key cartesian product size 4735 is greater than maximum 100">,`

We have increased the variables max_clustering_key_restrictions_per_query and max_partition_key_restrictions_per_query to 10000 in order to test because the applications weren't working properly.

Materialized views were working fine after the upgrade, but when we run the applications:

`Jul 23 11:31:24 scylladb-0 scylla:  [shard 8] storage_proxy - exception during mutation write to xxxxxxxxxxxx: seastar::named_semaphore_timed_out (Semaphore timed out: smp_service_group#2 13->8 semaphore)
Jul 23 11:31:24 scylladb-0 scylla:  [shard 2] storage_proxy - exception during mutation write to xxxxxxxxxxxx: seastar::semaphore_timed_out (Semaphore timedout)
Jul 23 11:31:24 scylladb-0 scylla: message repeated 2 times: [  [shard 2] storage_proxy - exception during mutation write to xxxxxxxxxxxx: seastar::semaphore_timed_out (Semaphore timedout)]`

The application logs:

`Error from server: code=1200 [Coordinator node timed out waiting for replica nodes' responses] message="Operation timed out for xxxxxxxxxxxxxxxxxxxxxx.xxxxxxxxxxxxxxxxxxxxxx - received only 1 responses from 2 CL=QUORUM." info={'consistency': 'QUORUM', 'required_responses': 2, 'received_responses': 1}, retries=1 [crawl_page]`

And following with high latency (printscreen).

In my point of view, we need to adjust the applications according to the cluster resources and follow the best practices.

Thank you.

![printscreens1](https://user-images.githubusercontent.com/10270413/88275229-5094f700-ccdd-11ea-9fcd-55da16debefb.png)
